tools/lsp/preview.rs

```diff
@@ -69,9 +70,56 @@ struct PreviewState {
     selected: Option<element_selection::ElementSelection>,
     notify_editor_about_selection_after_update: bool,
     known_components: Vec<ComponentInformation>,
+    document_cache: Option<common::DocumentCache>,
 }
 thread_local! {static PREVIEW_STATE: std::cell::RefCell<PreviewState> = Default::default();}
 
+impl PreviewState {
+    fn refresh_document_cache(
+        &mut self,
+        path: &Path,
+        version: SourceFileVersion,
+        source_code: String,
+    ) {
+        let Some(dc) = self.document_cache.as_mut() else {
+            return;
+        };
+
+        let mut diag = BuildDiagnostics::default();
+        spin_on::spin_on(dc.documents.load_file(
```

Comment:
could you add a comment explaining why it's Ok to call spin_on there?
Are you sure this doesn't do async calls?

Maybe instead of spin_on you could just poll the future once and ensure it's Ready.

You could also make this actually async since you are in the preview thread you can use slint::spawn_locale